pkg/distributor/http_server.go

```diff
@@ -35,11 +35,14 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if _, err := d.Push(r.Context(), &req); err != nil {
-		if httpResp, ok := httpgrpc.HTTPResponseFromError(err); ok {
-			level.Error(logger).Log("msg", "push error", "err", err)
+		level.Error(logger).Log("msg", "push error", "err", err)
+		if err == errIngestionRateLimitExceeded {
```

Comment:
Would it be better to have this 'thrown' as a `httpgrpc.Errorf` to begin with, as in https://github.com/weaveworks/cortex/blob/master/pkg/ingester/user_state.go#L203?

That was we don't end up with a big switch statement here, and you can provide more details in the error message (such as the observed rate).